### ISSUE TITLE
Wine system tray integration

### ISSUE THREAD

**Issue by bernharl** [Alexays/Waybar#1127]
Is is possible to integrate Wine system tray icons in Waybar's libappindicator tray? Right now all Wine trays open as separate windows in Sway.

**Comment by ErikReider** [comment]
A workaround would be to run `xembedsniproxy` which can be installed through the AUR (`xembed-sni-proxy-git`). It's a bit buggy where (at least on my setup) it displays a black box in the middle of the screen with the popup displaying on my left most monitor...

**Comment by eternal-sorrow** [comment]
FWIW there is a wine patch for SNI tray implementation, currently in the process of approval by upstream.

https://gitlab.winehq.org/wine/wine/-/merge_requests/2808

**Comment by neptuwunium** [comment]
Linked patch above is still not merged, also xembed-sni-proxy relies on KDE Plasmax modules (a bunch of distros are staging to remove it along with Plasma 5, as well?) Gentoo, Debian and Fedora package it either. Only AUR which still targets Plasma 5.

**Comment by corigne** [comment]
Yeah, the only workaround I could find for this was using the plasma-workspace AUR installer as xembed-sni-proxy is no longer available outside of plasma-workspace as a standalone tool unless you're willing to build it. 

I reached out to the author of xembed-sni-proxy via GH to ask if any considerations are required to ship [the origina code now merged into the plasma-workspace tree](https://phabricator.kde.org/source/plasma-workspace/browse/apol%252Fkdedkcmtosession/xembed-sni-proxy/) as a standalone tool while we wait for Wine and other apps to adopt the newer protocol.

A workaround for waybar would be good, a standalone xembed proxy slightly better, but in the end pushing for proper support for libappindicator tray icons in Wine and other applications would be better.

**Comment by BLumia** [comment]
> Yeah, the only workaround I could find for this was using the plasma-workspace AUR installer as xembed-sni-proxy is no longer available outside of plasma-workspace as a standalone tool unless you're willing to build it.

I have a standalone fork of `xembed-sni-proxy` btw: https://github.com/BLumia/xembed-sni-proxy

It's simply a standalone version of the xembed-sni-proxy extracted from plasma-framework. While it only serve for my testing purpose (I only need to run it directly from my build-tree without it actually "install" to somewhere system-level), feel free to try if it's useful for your use-case :)

**Comment by corigne** [comment]
> > Yeah, the only workaround I could find for this was using the plasma-workspace AUR installer as xembed-sni-proxy is no longer available outside of plasma-workspace as a standalone tool unless you're willing to build it.
> 
> I have a standalone fork of `xembed-sni-proxy` btw: https://github.com/BLumia/xembed-sni-proxy
> 
> It's simply a standalone version of the xembed-sni-proxy extracted from plasma-framework. While it only serve for my testing purpose (I only need to run it directly from my build-tree without it actually "install" to somewhere system-level), feel free to try if it's useful for your use-case :)

Thank you for working on an alternative. Unfortunately, I am unable to build your forked proxy to test it due to a cmake error. 

In either case, Hyprland (or some other process) is messing with the  with the XWayland helper windows used to render the drop-down menu, forcing them to render in the top-left of the current workspace instead of wherever the tray icon is.

<img width="452" height="441" alt="Image" src="https://github.com/user-attachments/assets/6edc92e9-de43-47f9-ab31-db62726c3183" />

**Comment by BLumia** [comment]
> Thank you for working on an alternative. Unfortunately, I am unable to build your forked proxy to test it due to a cmake error.

I cannot offer any help if you don't include the error detail. To avoid make this issue off-topic, If you are interested in a fix of my forked version of `xembed-sni-proxy`, you can create an issue to my fork repo with the complete build log (or at least include the log related to the error) so I can look into :)

**Comment by Alexays** [comment]
Closing — this was resolved/answered in the thread (environment, upstream, or a library limitation — not a Waybar bug still open). Feel free to reopen if you're still stuck. 🙏
